File: app/src/main/java/com/galib/natorepbs2/ui/ElectricityBillFragment.java
package com.galib.natorepbs2.ui;

import android.os.Bundle;

import androidx.databinding.DataBindingUtil;
import androidx.fragment.app.Fragment;
import androidx.navigation.Navigation;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.galib.natorepbs2.R;
import com.galib.natorepbs2.constants.Selectors;
import com.galib.natorepbs2.constants.URLs;
import com.galib.natorepbs2.databinding.FragmentElectricityBillBinding;

public class ElectricityBillFragment extends Fragment {

    public ElectricityBillFragment() {
        // Required empty public constructor
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        FragmentElectricityBillBinding binding = DataBindingUtil.inflate(inflater, R.layout.fragment_electricity_bill, container,false);
        binding.setPageTitle(getString(R.string.menu_electricity_bill));
        binding.setElectricityTariff(getString(R.string.menu_electricity_tariff));
        binding.setSmsBill(getString(R.string.menu_sms_electricity_bill));
        binding.setBillCollectionBank(getString(R.string.menu_bill_collection_bank));
        binding.setBillFromHome(getString(R.string.menu_bill_from_home));
        binding.setBillCalculator(getString(R.string.menu_electricity_bill_calculator));
        binding.setFragment(this);
        binding.setLifecycleOwner(getActivity());
        return binding.getRoot();
    }
    public void onClick(View v){
        //Navigation.findNavController(v).navigate(R.id.action_main_to_aboutUsFragment);
        int id = v.getId();
        if (id == R.id.bill_from_home_btn) {
            Navigation.findNavController(v).navigate(R.id.action_electricityBillFragment_to_billFromHomeFragment);
        } else if(id == R.id.electricity_tarriff_btn){
            ElectricityBillFragmentDirections.ActionElectricityBillFragmentToWebViewFragment action = ElectricityBillFragmentDirections.actionElectricityBillFragmentToWebViewFragment(getString(R.string.menu_electricity_tariff), URLs.BASE + URLs.TARIFF, null, Selectors.TARIFF);
            Navigation.findNavController(v).navigate(action);
        }
    }
}

File: app/src/main/java/com/galib/natorepbs2/ui/ElectricityBillFragment.kt
package com.galib.natorepbs2.ui

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.fragment.app.Fragment
import androidx.navigation.NavDirections
import androidx.navigation.fragment.findNavController
import com.galib.natorepbs2.R
import com.galib.natorepbs2.databinding.FragmentElectricityBillBinding
import com.galib.natorepbs2.utils.Utility

class ElectricityBillFragment : Fragment(),MenuOnClickListener {

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        val binding = DataBindingUtil.inflate<FragmentElectricityBillBinding>(
            inflater,
            R.layout.fragment_electricity_bill,
            container,
            false
        )
        binding.pageTitle = getString(R.string.menu_electricity_bill)
        binding.adapter = MenuAdapter(requireContext(),this, getMenuList())
        binding.lifecycleOwner = activity
        return binding.root
    }

    fun getMenuList():MutableList<String>{
        val list : MutableList<String> = ArrayList()
        list.add(getString(R.string.menu_bill_collection_bank))
        list.add(getString(R.string.menu_bill_from_home))
        list.add(getString(R.string.menu_electricity_tariff))
        list.add(getString(R.string.menu_electricity_bill_calculator))
        list.add(getString(R.string.menu_sms_electricity_bill))
        return list
    }

    override fun menuOnClick(menuText: String) {
        var action: NavDirections? = null
        when(menuText){
            getString(R.string.menu_bill_collection_bank) -> action = ElectricityBillFragmentDirections.actionElectricityBillFragmentToBankInformationFragment()
            getString(R.string.menu_bill_from_home) -> action = ElectricityBillFragmentDirections.actionElectricityBillFragmentToBillFromHomeFragment()
            getString(R.string.menu_electricity_tariff) -> action = ElectricityBillFragmentDirections.actionElectricityBillFragmentToWebViewFragment(getString(R.string.menu_electricity_tariff), null, Utility.getTariffHtml(requireContext().assets), null)
            getString(R.string.menu_electricity_bill_calculator) -> action = ElectricityBillFragmentDirections.actionElectricityBillFragmentToBillCalculatorFragment()
            getString(R.string.menu_sms_electricity_bill) -> action = ElectricityBillFragmentDirections.actionElectricityBillFragmentToBillMessageFragment()
        }
        if(action != null)
            findNavController().navigate(action)
    }
}